### PR TITLE
initial style guide

### PR DESCRIPTION
**Pull request by ericvw**
Currently addresses the following:
- files
- indentation
- line length

This also addresses #41 as start.

### CODE REVIEW

File: doc/style-guide.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    IDE/editors.  Most editors have configuration options to change tabs to spaces.
+
+**[top](#table-of-contents)**
+
+## Line Length
+  - Lines should not be longer than 100 characters.
+  - When a statement runs over 100 characters on a line, it should be broken by either a comma or
+    operator.
+  - Long strings at the end of statements that go over 100 characters is acceptable on the basis
+    of readability.  Please your own judgement of what is acceptable for being over the 100
```
Comment (ericlu88) [inline doc/style-guide.md]: Please **use** your own judgement...

File: doc/style-guide.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  1. [Introduction](#introduction)
+  1. [Files](#files)
+  1. [Indentation](#indentation)
+  1. [Line Length](#line-length)
+
+
+## Introduction
+
+When developing software in collaboration with many different individuals, the value of the of the
```
Comment (yupinghu) [inline doc/style-guide.md]: Delete one of the "of the".

File: doc/style-guide.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+software is directly affected by the quality of the codebase.  Consistent coding conventions are
+important when the code is developed over many years and handled/seen by others.  Consistent
+conventions lend itself to easier readability and understanding of the codebase for newcomers,
```
Comment (yupinghu) [inline doc/style-guide.md]: "Consistent conventions lend themselves" (or make the subject singular?)

File: doc/style-guide.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+## Indentation
+  - The unit of indentation is four spaces.
+  - **Never use tabs**, this causes problems with differing interpretations by different
+    IDE/editors.  Most editors have configuration options to change tabs to spaces.
+
+**[top](#table-of-contents)**
+
+## Line Length
+  - Lines should not be longer than 100 characters.
+  - When a statement runs over 100 characters on a line, it should be broken by either a comma or
```
Comment (yupinghu) [inline doc/style-guide.md]: Is there better wording to clarify that the wrap happens after the comma/operator?

**Comment by ericvw** [comment]
Updated based on previous feedback.  Note that I will need to squash everything before merging.

**Comment by ericvw** [comment]
Updated based on latest feedback.

**Pull request merged**